src/deluge/gui/ui/browser/browser.cpp

```diff
@@ -304,10 +295,12 @@ Error Browser::readFileItemsForFolder(char const* filePrefixHere, bool allowFold
 		audioFileManager.loadAnyEnqueuedClusters();
 		FilePointer thisFilePointer;
 
-		result = f_readdir_get_filepointer(&staticDIR, &staticFNO, &thisFilePointer); /* Read a directory item */
+		std::tie(staticFNO, thisFilePointer) = D_TRY_CATCH(dir.read_and_get_filepointer(), error, {
```

Comment:
Could we use an actual try/catch now?

Reply:
nope, the FatFS interface uses std::expected, which is what D_TRY/D_TRY_CATCH work with. I honestly prefer std::expected to exceptions when possible anyways, closer to Rust's Result and it doesn't rely on the weird type matching